src/coreclr/jit/compiler.h

```diff
@@ -7072,6 +7072,7 @@ class Compiler
                 IntegralRange u2;
             };
         } op2;
+        bool vnBased : 1;
```

Comment:
I don't think you need to increase the size of `AssertionDsc` for this, you can just make `optAssertionKind` a `short`-backed enum and put `vnBased` after it:
```cs
struct Compiler::AssertionDsc /* size 48 @ C:\Users\Accretion\source\dotnet\runtime\src\coreclr\jit\compiler.h:7033 */
{
/* 0000:04 */ Compiler::optAssertionKind assertionKind;
/* 0004:16 */ Compiler::AssertionDsc::AssertionDscOp1 op1;
/* 0020:04 */ 4 bytes padding
/* 0024:24 */ Compiler::AssertionDsc::AssertionDscOp2 op2;
};
```
Notably, the `AssertionDsc` structure is bigger than it needs to be in other ways as well, both `optOp1Kind` and `optOp2Kind` enums don't need to be 4 bytes large, and that 4 byte padding in the middle should not exist.

Reply:
That's true. Will change it.